src/ProxyServer/ProxyServer.swift

```diff
@@ -29,6 +29,8 @@ open class ProxyServer: NSObject, TunnelDelegate {
     open var observer: Observer<ProxyServerEvent>?
 
     var tunnels: TunnelArray = []
+    
+    private let accessQueue = DispatchQueue(label: "NEKit.ProxyServer.TunnelArrayAccess", attributes: .concurrent)
```

Comment:
Ehh.. If this is a concurrent queue, then it won't protect any race condition.
There is `Atomic` provided just for this usage.